tasks_management/signals/on_task_resolve.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from core.forms import User
+from tasks_management.apps import TasksManagementConfig
+from tasks_management.models import Task
+from tasks_management.services import TaskService
+
+logger = logging.getLogger(__name__)
+
+
+def resolve_task_all(_task, _user):
+    if 'FAILED' in _task.business_status.values():
+        TaskService(_user).complete_task({"id": _task.id, 'failed': True})
+    if sum(map('APPROVED'.__eq__, _task.business_status.values())) == _task.task_group.taskexecutor_set.count():
+        TaskService(_user).complete_task({"id": _task.id})
```

Comment:
I think that we could improve readability of  part with sum(map(...)). 
Can you check if 
```suggestion
def resolve_task_all(_task, _user):
    # Check for any 'FAILED' status
    if 'FAILED' in _task.business_status.values():
        TaskService(_user).complete_task({"id": _task.id, 'failed': True})

    # Check if all statuses are 'APPROVED'
    all_approved = all(status == 'APPROVED' for status in _task.business_status.values())
    approved_count_match = len(_task.business_status.values()) == _task.task_group.taskexecutor_set.count()

    if all_approved and approved_count_matche:
        TaskService(_user).complete_task({"id": _task.id})

```
would work?

Reply:
I've added names for the left and right side of the check, but i left the sum(map(...)) solution, as I think it's better. I also fixed missign is_deleted check.